### PR TITLE
Fix/getblobbers

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
### Changes
- Fixes pagination handling logic

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/getblobbers  |
| zboxcli  | fix/get-blobber  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/getblobbers  |
| zboxcli  | staging  |
| zwalletcli  | fix/get-blobber  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/getblobbers  |
| zboxcli  | staging  |
| zwalletcli  | fix/get-blobber  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/getblobbers  |
| zboxcli  | staging  |
| zwalletcli  | fix/get-blobber  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: zcncore/wallet_base.go [hunk 1]
```diff
@@ -980,13 +980,30 @@ func GetStakePoolUserInfo(clientID string, cb GetInfoCallback) (err error) {
 }
 
 // GetBlobbers obtains list of all active blobbers.
-func GetBlobbers(cb GetInfoCallback) (err error) {
-	if err = CheckConfig(); err != nil {
+func GetBlobbers(cb GetInfoCallback, limit, offset int, options ...bool) {
```
Comment (peterlimg) [inline zcncore/wallet_base.go:1012]: We could do the API better by using functional option for all the limit, offet, and active options, so when by default we will get all active blobbers, with `GetBlobbers(cb GetInfoCallback)`, and if we only need subset with pagination, we then call it like `GetBlobbers(cb GetInfoCallback, WithOffset(100), WithLimit(20))` and if needed to get all by `WithAll()`. 

Also, we should use the `cb GetInfoCallback)` internal, the API callers should not be bothered to setup the callback function, they should only call this API be like `GetBlobber(ctx, options...) (results, error)`. 

Anyway, that's the direction we need to refactor later, I'll approve all related PRs as these changes can be done in new PRs.

**Review by peterlimg — APPROVED** [review]

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | fix/getblobbers  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: zcncore/wallet_base.go (hunk 1 above)
Comment (boddumanohar) [reply zcncore/wallet_base.go:1012]: https://github.com/0chain/gosdk/issues/660

thanks for the feedback @peterlimg . Created an issue to track this.

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**